src/run.ts

```diff
@@ -1,147 +1,99 @@
+        console.log(`Failed to delete secret with statusCode: ${response?.statusCode}`)
+        console.log(response?.body?.metadata)
     }
+    console.log('Deleting secret:')
+    console.log(deleteSecretResponse?.response?.body)
 
-    args.push('--docker-email', email);
-    return args;
-}
 
-function getGenericSecretArguments(secretName: string): string[] {
-    const secretArguments = core.getInput('arguments');
-    const parsedArgument = fromLiteralsToFromFile(secretArguments);
-    let args = ['create', 'secret', 'generic', secretName];
-    args.push(...argStringToArray(parsedArgument));
-    return args;
-}
+    const secret = await buildSecret(secretName, namespace)
 
-/**
- * Takes a valid kubectl arguments and parses --from-literal to --from-file
- * @param secretArguments 
- */
-export function fromLiteralsToFromFile(secretArguments: string): string {
-    const parsedArgument = secretArguments.split("--").reduce((argumentsBuilder, argument) => {
-        if (argument && !argument.startsWith("from-literal=")) {
-            argumentsBuilder = argumentsBuilder.trim() + " --" + argument;
-        } else if (argument && argument.startsWith("from-literal=")) {
-            const command = argument.substring("from-literal=".length);
-            /* The command starting after 'from-literal=' contanis a 'key=value' format. The secret itself might contain a '=', 
-            Hence the substring than a split*/
-            if (command.indexOf("=") == -1) throw new Error('Invalid from-literal input. It should contain a key and value');
-            const secretName = command.substring(0, command.indexOf("=")).trim();
-            const secretValue = command.substring(command.indexOf("=") + 1).trim();
-            //Secret with spaces will be enclosed in quotes -> "secret "
-            if (secretValue && secretValue.indexOf("\"") == 0 && secretValue.lastIndexOf("\"") == secretValue.length - 1) {
-                const secret = secretValue.substring(1, secretValue.lastIndexOf("\""));
-                argumentsBuilder += " --from-file=" + fileUtility.createFile(secretName, secret, true);
-            } else {
-                const secret = secretValue.substring(0, secretValue.indexOf(" ") == -1 ? secretValue.length : secretValue.indexOf(" "));
-                argumentsBuilder += " --from-file=" + fileUtility.createFile(secretName, secret, true);
-            }
-        }
-        return argumentsBuilder;
-    });
-    return parsedArgument.trim();
-}
+    let result;
 
-function checkClusterContext() {
-    if (!process.env["KUBECONFIG"]) {
-        throw new Error('Cluster context not set. Use k8s-set-context/aks-set-context action to set cluster context');
+    try {
+        result = await api.createNamespacedSecret(namespace, secret)
+    } catch (e) {
+        let response = e?.response
+
+        console.log(`Failed to create secret with statusCode: ${response?.statusCode}`)
+        console.log(response?.body)
```

Comment:
i don't think you need both the console.log here and the core.setFailed. Instead of using console.log I think core.debug is better

Reply:
it looks like core.debug isn't visible unless you set a secret to see debug logs. is that an acceptable default behavior?